packages/mux-player/src/index.jsx

```diff
@@ -0,0 +1,413 @@
+        hasAirPlay={hasAirPlay}
+        onAirPlaySelected={() => {
+          muxVideoRef.current?.webkitShowPlaybackTargetPicker?.();
+        }}
+        streamType={streamType}
+        playerSize={playerSize}
+      />
+    </MediaController>
+  );
+};
+
+const VideoAttributes = {
+  ENV_KEY: "env-key",
+  DEBUG: "debug",
+  PLAYBACK_ID: "playback-id",
+  METADATA_URL: "metadata-url",
+  PREFER_MSE: "prefer-mse",
+  METADATA_VIDEO_ID: "metadata-video-id",
+  METADATA_VIDEO_TITLE: "metadata-video-title",
+  METADATA_VIEWER_USER_ID: "metadata-viewer-user-id",
+  BEACON_DOMAIN: "beacon-domain",
+  TYPE: "type",
+  STREAM_TYPE: "stream-type",
+  START_TIME: "start-time",
+};
+
+const VideoAttributeNameValues = Object.values(VideoAttributes);
+const SMALL_BREAKPOINT = 700;
+
+function renderMuxPlayer(el) {
+  let mediaControllerRef = {};
+  let muxVideoRef = {};
+  let playerSize = getPlayerSize(el);
+
+  function getPlayerSize(el) {
+    const muxPlayerRect = el.getBoundingClientRect();
+    return muxPlayerRect.width < SMALL_BREAKPOINT
+      ? MediaChromeSizes.SM
+      : MediaChromeSizes.LG;
+  }
+
+  function renderToShadowRoot() {
+    console.log("RENDER", el.playbackId);
+    render(
+      <MuxPlayer
+        playbackId={el.playbackId}
+        mediaControllerRef={mediaControllerRef}
+        muxVideoRef={muxVideoRef}
+        playerSize={playerSize}
+      />,
+      el.shadowRoot
+    );
+  }
+
+  renderToShadowRoot();
+
+  window.addEventListener("resize", () => {
+    if (playerSize != getPlayerSize(el)) {
+      playerSize = getPlayerSize(el);
+      renderToShadowRoot();
+    }
+  });
+}
+
+class MuxPlayerElement extends HTMLElement {
+  static get observedAttributes() {
+    return [...VideoAttributeNameValues];
+  }
+
+  constructor() {
+    super();
+
+    this.attachShadow({ mode: "open" });
+    renderMuxPlayer(this);
+
+    // Initialize all the attribute properties
+    Array.prototype.forEach.call(this.attributes, (attrNode) => {
+      this.attributeChangedCallback(attrNode.name, null, attrNode.value);
+    });
+
+    if (this.video) this.video.hls.config.maxMaxBufferLength = 2;
+  }
+
+  get video() {
+    return this.shadowRoot.querySelector("mux-video");
+  }
+
+  getVideoAttribute(name) {
+    return this.video ? this.video.getAttribute(name) : this.getAttribute(name);
+  }
+
+  get envKey() {
+    return this.getVideoAttribute(VideoAttributes.ENV_KEY);
+  }
+
+  get debug() {
+    return this.getVideoAttribute(VideoAttributes.DEBUG);
+  }
+
+  get playbackId() {
+    return this.getVideoAttribute(VideoAttributes.PLAYBACK_ID);
+  }
+
+  get streamType() {
+    return this.getVideoAttribute(VideoAttributes.STREAM_TYPE);
+  }
+
+  get envKey() {
+    return this.getVideoAttribute(VideoAttributes.ENV_KEY);
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (VideoAttributeNameValues.includes(attrName)) {
+      this.video?.attributeChangedCallback(attrName, oldValue, newValue);
```

Comment:
I think you want this to be `this.video.setAttribute(attrName, newValue);`

Reply:
Also I think this will have edge cases if you're concerned about this.video not existing.